File: src/app/Models/cartItem.ts
import { identifierName } from "@angular/compiler";

export class CartItem {
    productId: string;
    customerId: string;  
    quantityInCart: number; 
    selectedForBuying: boolean;

    constructor(
        productId: string, 
        customerId: string,
        quantityInCart: number,
        selectedForBuying: boolean,
    ) {
        this.productId = productId;
        this.customerId = customerId;
        this.quantityInCart = quantityInCart;
        this.selectedForBuying = selectedForBuying;
    }
}

File: GoServer/Model/cartItem.go
package Model

type CartItem struct {
	ProductId        	    string      `json:"productId" bson:"productId"`
	CustomerId        	  string      `json:"customerId" bson:"customerId"`
	QuantityInCart 		    int 		    `json:"quantityInCart" bson:"quantityInCart"`
	SelectedForBuying 	  bool		    `json:"selectedForBuying" bson:"selectedForBuying"`
}
